src/tools/cursors/MouseCursor.js

```diff
@@ -0,0 +1,77 @@
+import toolColors from '../../stateManagement/toolColors.js';
+import { modules } from '../../store/index.js';
+
+const cursorModule = modules.cursor;
+
+export default class MouseCursor {
+  constructor(iconGroupString, options) {
+    this.iconGroupString = iconGroupString;
+    this.options = Object.assign(
+      {},
+      cursorModule.getters.defaultOptions(),
+      options
+    );
+  }
+
+  get iconSVG() {
+    const svgString = this._generateIconSVGString();
+
+    return new Blob([svgString], { type: 'image/svg+xml' });
+  }
+
+  get iconWithPointerSVG() {
+    const svgString = this._generateIconWithPointerSVGString();
+
+    return new Blob([svgString], { type: 'image/svg+xml' });
+  }
```

Comment:
If we only update this property's value when the `svgString` or `options` change, it might be a bit more performant.

Reply:
Hmm yeah, maybe I'll put some caching in.

Reply:
Hmm, how do we check if the tool color has changed? 🤔

Reply:
:l 
...
Shoot.

We could add an event? Or, if we're only setting the SVG cursor when the tool is activated / made primary, then we can use whatever color at that point in time and call it good enough?

Reply:
> Or, if we're only setting the SVG cursor when the tool is activated / made primary, then we can use whatever color at that point in time and call it good enough?

Maybe I misunderstand, but isn't that what we are already doing? I personally thinking adding another event is a little overkill, how often is the active color changed on run time? I've generally seen it as more of a set-and-forget. Even if the active color is changed during run time, it will at least refresh on the next tool activation?

Reply:
Ahh! It [looks like we are](https://github.com/cornerstonejs/cornerstoneTools/pull/887/files/cbe4317c1a774780e0a1ea0f9a72b9ee74749d33#diff-0c9bc112b60ea20dbbd1f7953c4c2c29R90). I had this weird thought that we were setting the cursor on every render (oops).

You can ignore my comments then. I was trying to optimize for a situation/issue that we don't have.